cleanup_image.sh

```diff
@@ -11,6 +12,14 @@ mv "${mount}/etc/_resolv.conf" "${mount}/etc/resolv.conf"
 
 [[ -f "${mount}/tmp/commands.sh" ]] && rm "${mount}/tmp/commands.sh"
 if [[ -d "${mount}" ]]; then
+    if [[ "${optimize}x" != "x" && "${optimize}x" != "nox" ]]; then
+        if [[ -d "${mount}/boot" ]]; then
+            echo "Zero-filling unused blocks on /boot filesystem..."
+            cat /dev/zero >"${mount}/boot/zero.fill" 2>/dev/null; sync; rm -f "${mount}/boot/zero.fill"
+        fi
+        echo "Zero-filling unused blocks on / filesystem..."
+        cat /dev/zero >"${mount}/zero.fill" 2>/dev/null; sync; rm -f "${mount}/zero.fill"
```

Comment:
Wondering if we should optimize this with dd instead of cat.

Reply:
Hmmm...
Time wise, `cat /dev/zero` is pretty efficient, at least in the GitHub qemu emulation, and has negligible impact on action execution time.
And I wonder how dd could do the job: `dd if=/dev/zero of=...` would erase the whole destination filesystem...

Reply:
We should measure the thing of course. I admit I haven't checked how long cat did take.
The syntax would be:
```
dd if=/dev/zero of=${mount}/zero.fill bs=...
```
with various values of bs (e.g. 1m)

Reply:
> dd if=/dev/zero of=${mount}/zero.fill bs=...

Ah... got it. But bs should not be arbitrary: we would need to parse the output of `df` on given fs to set it from the available space. Do we need such complexity?

Reply:
Apparently, cat is considered as fast or faster than dd on modern Linux, so this is a bad premature optimization suggestion.
Where did you actually run this code? Couldn't find it in your pynab fork actions. I am wondering how long this cat command line is, as if it's cheap enough it could be enabled by default, even without running PiShrink this definitely will reduce compressed size of image.
Also, I believe the cat command line is not run inside qemu but natively, qemu chroot being only used for the provided commands script.

Reply:
> Where did you actually run this code? Couldn't find it in your pynab fork actions. I am wondering how long this cat command line is
You have an example here: https://github.com/f-laurens/pynab/actions/runs/967248905
It uses an [instrumented version of cleanup.sh](https://github.com/f-laurens/arm-runner-action/blob/79fd97b380ee2889937030667e67c0397c8b4a0f/cleanup_image.sh) providing the timinfg figures you wanted:

- RaspiOS:
```
Zero-filling (cat) unused blocks on /boot filesystem...
real	0m0.624s
user	0m0.000s
sys	0m0.624s
Zero-filling (dd 209299k) unused blocks on /boot filesystem...
real	0m0.727s
user	0m0.000s
sys	0m0.727s
Zero-filling (cat) unused blocks on / filesystem...
real	0m6.770s
user	0m0.016s
sys	0m1.862s
Zero-filling (dd 1993828k) unused blocks on / filesystem...
real	0m9.301s
user	0m0.006s
sys	0m3.250s
```
- DietPi:
```
Zero-filling (cat) unused blocks on /boot filesystem...
real	0m0.754s
user	0m0.004s
sys	0m0.687s
Zero-filling (dd 207593k) unused blocks on /boot filesystem...
real	0m0.703s
user	0m0.000s
sys	0m0.702s
Zero-filling (cat) unused blocks on / filesystem...
real	0m7.961s
user	0m0.000s
sys	0m1.482s
Zero-filling (dd 1514904k) unused blocks on / filesystem...
real	0m6.070s
user	0m0.000s
sys	0m2.477s
```
It's a tie between cat and dd, so my vote would be to keep the simple cat-based implementation.
 > as if it's cheap enough it could be enabled by default,

Reply:
> even without running PiShrink this definitely will reduce compressed size of image.

Indeed, for Pynab images, we might do the the xz compression ourselves.
I'd even be in favor of getting rid of PiShrink altogether, since:

- it has no reliable release/tagging mechanism;
- it does useless/obsolete stuff when not given the -s option;
- this would eliminate the arguments about using the -p option or not ;-).

Some stuff we might want to keep/steal (add it to arm-runner cleanup step when optimize_image is set) from PiShrink is its resizing of the root partition to minsize + some_free_space, so that resulting image may be installed on small SD cards (cheapo 4GB cards, if such things still exist ;-).

Reply:
> I believe we should enable it by default.

What do you mean exactly?
- Suppress `optimize_image`parameter and have zero-fill all the time?
As written above, I'd not be in favor of that.
- Or just change default for `optimize_image` from `no` to `yes`?
I'd favor that.

Reply:
I also favor just changing default for new `optimize_image` option.

Reply:
Fine. I pushed the amended PR.
I guess it is now ripe for merging, and bumping the release tag accordingly.